File: SDKs/beezup.com/2.0/python/src/sdk/models/shared/reportfilter.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import reportfilterlinks as shared_reportfilterlinks
from ..shared import reportfilterparameters as shared_reportfilterparameters


@dataclass_json
@dataclasses.dataclass
class ReportFilter:
    parameters: shared_reportfilterparameters.ReportFilterParameters = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('parameters') }})
    report_filter_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('reportFilterId') }})
    report_filter_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('reportFilterName') }})
    links: Optional[shared_reportfilterlinks.ReportFilterLinks] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('links') }})
    

File: SDKs/beezup.com/2.0/typescript/src/sdk/models/shared/reportfilter.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ReportFilterLinks } from "./reportfilterlinks";
import { ReportFilterParameters } from "./reportfilterparameters";



export class ReportFilter extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=links" })
  links?: ReportFilterLinks;

  @SpeakeasyMetadata({ data: "json, name=parameters" })
  parameters: ReportFilterParameters;

  @SpeakeasyMetadata({ data: "json, name=reportFilterId" })
  reportFilterId: string;

  @SpeakeasyMetadata({ data: "json, name=reportFilterName" })
  reportFilterName: string;
}
